metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/profiler.py

```diff
@@ -78,39 +81,57 @@ def generate_partition_profiler_query(
         partition = table.max_partition_id
         if partition:
             partition_where_clause: str
-            logger.debug(f"{table} is partitioned and partition column is {partition}")
-            try:
-                (
-                    partition_datetime,
-                    upper_bound_partition_datetime,
-                ) = self.get_partition_range_from_partition_id(
-                    partition, partition_datetime
-                )
-            except ValueError as e:
-                logger.error(
-                    f"Unable to get partition range for partition id: {partition} it failed with exception {e}"
-                )
-                self.report.invalid_partition_ids[f"{schema}.{table}"] = partition
-                return None, None
-
-            if table.time_partitioning.type_ in ("DAY", "MONTH", "YEAR"):
-                partition_where_clause = "{column_name} BETWEEN DATE('{partition_id}') AND DATE('{upper_bound_partition_id}')".format(
-                    column_name=table.time_partitioning.field,
-                    partition_id=partition_datetime,
-                    upper_bound_partition_id=upper_bound_partition_datetime,
-                )
-            elif table.time_partitioning.type_ in ("HOUR"):
-                partition_where_clause = "{column_name} BETWEEN '{partition_id}' AND '{upper_bound_partition_id}'".format(
-                    column_name=table.time_partitioning.field,
-                    partition_id=partition_datetime,
-                    upper_bound_partition_id=upper_bound_partition_datetime,
-                )
+
+            if not table.time_partitioning:
+                partition_column: Optional[BigqueryColumn] = None
+                for column in table.columns:
+                    if column.is_partition_column:
+                        partition_column = column
```

Comment:
can there be multiple partition columns?

Reply:
BigQuery only supports partitioning on one column.